activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -262,8 +283,20 @@ def active_connection?
       # This method only works for connections that have been obtained through
       # #connection or #with_connection methods, connections obtained through
       # #checkout will not be automatically released.
-      def release_connection(owner_thread = ActiveSupport::IsolatedExecutionState.context)
-        if conn = @thread_cached_conns.delete(owner_thread)
+      def release_connection(existing_lease = nil)
+        lease = if existing_lease
+          if @thread_connection_leases.delete_pair(ActiveSupport::IsolatedExecutionState.context, existing_lease)
+            # The connection may have been reaped, so we shouldn't try to check it back in
+            # if it's no longer in use.
```

Comment:
Is this newly true, or just added caution? AFAIK the reaper should never touch a connection that's leased by a still-running thread

Reply:
IIRC I experienced it in the test suite :/ need to dig more.

Reply:
Seems like it's no longer failing, so I removed it.